pdns/dnsdist-tcp.cc

```diff
@@ -341,207 +339,55 @@ void tcpClientThread(int pipefd)
           break;
         }
 
-        bool ednsAdded = false;
-        bool ecsAdded = false;
         /* allocate a bit more memory to be able to spoof the content,
            or to add ECS without allocating a new buffer */
-        queryBuffer.resize(qlen + 512);
+        queryBuffer.resize((static_cast<size_t>(qlen) + 512) < 4096 ? (static_cast<size_t>(qlen) + 512) : 4096);
```

Comment:
`std::min` ?

Reply:
It makes complete sense but this code goes away in #7559 :)